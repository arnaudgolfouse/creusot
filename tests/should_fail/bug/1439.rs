extern crate creusot_contracts;
use creusot_contracts::*;

pub enum Empty {}

#[requires(match empty { })]
pub fn foo(empty: Empty) {}
