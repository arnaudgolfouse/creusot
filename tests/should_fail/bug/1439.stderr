error: Empty matches are forbidden in Pearlite, because Why3 types are always inhabited.
 --> 1439.rs:6:12
  |
6 | #[requires(match empty { })]
  |            ^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

